### PR TITLE
Support allure-pytest for generating reports

### PR DESCRIPTION
**Pull request by YuliiaBashko**
Signed-off-by: Yuliia Bashko <yuliiax.bashko@intel.com>

#### Why I did it
To provide full picture of test regression results e.g what features have been covered, where defects are clustered and how the timeline of execution looks like
#### How I did it
Added allure-pytest plugin to sonic-mgmt Dockerfile
#### How to verify it
Check if plugin is installed in container with `pip show  allure-pytest`
#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012

### CODE REVIEW

**Review by akokhan — APPROVED** [review]

File: dockers/docker-sonic-mgmt/Dockerfile.j2 [hunk 1]
```diff
@@ -60,6 +60,7 @@ RUN pip install cffi==1.10.0 \
                 pysnmp==4.2.5 \
                 pytest-repeat \
                 pytest-html \
+                allure-pytest==2.8.22 \
```
Comment (lguohan) [inline dockers/docker-sonic-mgmt/Dockerfile.j2:63]: why not use the latest version?

Comment (YuliiaBashko) [reply dockers/docker-sonic-mgmt/Dockerfile.j2:63]: As sonic-mgmt framework is on python2, I choose the latest allure-pytest version that supports python2, another allure-pytest require higher version than python2.7

Comment (jleveque) [inline dockers/docker-sonic-mgmt/Dockerfile.j2:63]: Please keep list in alphabetical order.

**Review by jleveque — CHANGES_REQUESTED** [review]

File: dockers/docker-sonic-mgmt/Dockerfile.j2 (hunk 1 above)
Comment (YuliiaBashko) [reply dockers/docker-sonic-mgmt/Dockerfile.j2]: ok, fixed

File: dockers/docker-sonic-mgmt/Dockerfile.j2 [hunk 2]
```diff
@@ -33,7 +33,8 @@ RUN apt-get update && apt-get install -y build-essential \
                                          telnet \
                                          vim
 
-RUN pip install cffi==1.10.0 \
+RUN pip install allure-pytest==2.8.22 \
+		cffi==1.10.0 \
```
Comment (jleveque) [inline dockers/docker-sonic-mgmt/Dockerfile.j2:37]: Please align this line with the others (it appears there are tabs here when there should be spaces).

**Review by jleveque — CHANGES_REQUESTED** [review]

File: dockers/docker-sonic-mgmt/Dockerfile.j2 (hunk 2 above)
Comment (YuliiaBashko) [reply dockers/docker-sonic-mgmt/Dockerfile.j2]: fixed

**Review by yxieca — APPROVED** [review]

**Comment by ppikh** [comment]
Hi,

In case when we have "pip install allure-pytest==2.8.22" in line 36 - docker container will not build, it will raise error.

```
File "/tmp/easy_install-ph8nXg/setuptools_scm-6.0.1/src/setuptools_scm/config.py", line 6, in <module>
File "/tmp/easy_install-ph8nXg/setuptools_scm-6.0.1/src/setuptools_scm/utils.py", line 41
print(*k)
^
SyntaxError: invalid syntax

----------------------------------------
Command "python setup.py egg_info" failed with error code 1 in /tmp/pip-build-trOHiJ/allure-pytest/
```

We need to do command "pip install --upgrade setuptools" to prevent this error.
And we need to do "pip install --upgrade setuptools" at the end of Docker file, because this command may affect installation of another python libraries. Please check this commit: https://github.com/Azure/sonic-buildimage/pull/7665

**Comment by yxieca** [comment]
@jleveque  do you have further concerns?

**Comment by yxieca** [comment]
@YuliiaBashko can you check comments from @ppikh ? Did you see same failure?

**Comment by YuliiaBashko** [comment]
> @YuliiaBashko can you check comments from @ppikh ? Did you see same failure?

I rechecked my changes and received the same error like @ppikh provided. We should use #7665 to avoid this behavior.

```
 File "/tmp/easy_install-Euxy4z/setuptools_scm-6.0.1/src/setuptools_scm/utils.py", line 41
        print(*k)
              ^
    SyntaxError: invalid syntax
    

    ----------------------------------------

Command "python setup.py egg_info" failed with error code 1 in /tmp/pip-build-48qqwx/allure-pytest/
```

**Comment by yxieca** [comment]
> > @YuliiaBashko can you check comments from @ppikh ? Did you see same failure?
> 
> I rechecked my changes and received the same error like @ppikh provided. We should use #7665 to avoid this behavior.
> 
> ```
>  File "/tmp/easy_install-Euxy4z/setuptools_scm-6.0.1/src/setuptools_scm/utils.py", line 41
>         print(*k)
>               ^
>     SyntaxError: invalid syntax
>     
> 
>     ----------------------------------------
> 
> Command "python setup.py egg_info" failed with error code 1 in /tmp/pip-build-48qqwx/allure-pytest/
> ```
Thanks @YuliiaBashko for confirming! I'll merge #7665 then.

**Pull request closed without merging**